### PR TITLE
chore(fail-open): Update fail-open message body

### PR DESCRIPTION
**Pull request by zyannes**
PR checklist:

- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Changelog guidelines](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry)
- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: cli/src/semgrep/error_handler.py [hunk 1]
```diff
@@ -72,13 +73,17 @@ def send(self, exit_code: int) -> int:
             f"Sending to fail-open endpoint {state.env.fail_open_url} since fail-open is configured to {self.suppress_errors}"
         )
 
-        headers = self.payload.get("headers", {})
-        headers["User-Agent"] = str(state.app_session.user_agent)
-        self.payload["headers"] = headers
+        token = auth.get_token()
+        headers = {
+            "User-Agent": str(state.app_session.user_agent),
+            "Authorization": f"Bearer {token or ''}",
```
Comment (spencerdrak) [inline cli/src/semgrep/error_handler.py:79]: will this token be captured even if we fail early in the process?

File: cli/tests/unit/test_error_handler.py [hunk 2]
```diff
@@ -35,35 +50,38 @@ def test_send_nominal(error_handler, mocker: MockerFixture) -> None:
     exit_code = error_handler.send(FATAL_EXIT_CODE)
 
     expected_payload = {
-        "headers": {"User-Agent": expected_user_agent},
         "method": "get",
         "url": "https://semgrep.dev/api/agent/deployments/current",
     }
 
+    expected_headers = {
```
Comment (spencerdrak) [inline cli/tests/unit/test_error_handler.py:57]: Are these the headers as SQS sees them or as they are in the body of the requet?

File: cli/src/semgrep/error_handler.py (hunk 1 above)
Comment (zyannes) [reply cli/src/semgrep/error_handler.py:79]: Yes. The token is pulled from an env variable (most commonly for CI) or a `.semgrep/settings.yml` file. So regardless of the failure, the token should be available (or worst case just returns `None`)

File: cli/tests/unit/test_error_handler.py (hunk 2 above)
Comment (zyannes) [reply cli/tests/unit/test_error_handler.py:57]: These are actual headers that will be forwarded to the SQS.

**Review by brendongo — APPROVED** [review]

**Pull request merged**